server/src/main/java/org/opensearch/index/store/lockmanager/FileLockInfo.java

```diff
@@ -88,7 +90,11 @@ static String generateLockName(String fileToLock, String acquirerId) {
         }
 
         public static String getFileToLockNameFromLock(String lockName) {
-            String[] lockNameTokens = lockName.split(RemoteStoreLockManagerUtils.SEPARATOR);
+            if (lockName.endsWith(RemoteStoreLockManagerUtils.V1_LOCK_FILE_EXTENSION)) {
+                // this is the old lock file created for versions <= 2.10
+                return getFileToLockNameFromV1Lock(lockName);
+            }
+            String[] lockNameTokens = lockName.split(Pattern.quote(RemoteStoreLockManagerUtils.SEPARATOR));
```

Comment:
for this as well I will use the `lastIndexOf` instead of `split` as this will fail if metadata file name start using the same separator as lock manager. With `lastIndexOf` lock manager knows that it adds the info towards the end of any given file name so thats where it should look for. We can do same as for acquirer id.


```
private static String getFileToLockNameFromLock(String lockName) {
            // use proper separator for the lock file depending on the version it is created
          String lockSeparator = lockName.endsWith(RemoteStoreLockManagerUtils.V1_LOCK_FILE_EXTENSION) ? RemoteStoreLockManagerUtils.V1_LOCK_SEPARATOR : RemoteStoreLockManagerUtils.SEPARATOR;
        final int indexOfSeparator = lockName.lastIndexOf(lockSeparator) ;
        if (indexOfSeparator == INVALID_INDEX) {
                throw new IllegalArgumentException("Provided lock name: " + lockName + " is invalid with separator: " + lockSeparator);
        }
         return lockName.substring(0, indexOfSeparator);
}
```